show-device-location-using-indoor-positioning/src/main/java/com/esri/arcgismaps/sample/showdevicelocationusingindoorpositioning/MainActivity.kt

```diff
@@ -0,0 +1,350 @@
+    private fun setupIndoorsLocationDataSource(featureTable: FeatureTable) {
+        // positioningTable needs to be present
+        if (featureTable.tableName == "IPS_Positioning") {
+            val positioningFeatureTable = featureTable as ServiceFeatureTable
+            // when multiple entries are available, IndoorsLocationDataSource constructor function
+            // looks up the entry with the most recent date and takes this positioning data
+            // set up queryParameters to grab one result.
+            val dateCreatedFieldName =
+                getDateCreatedFieldName(positioningFeatureTable.fields)
+            if (dateCreatedFieldName == null) {
+                showError("The service table does not contain \"DateCreated\" fields.")
+                return
+            }
+            val queryParameters = QueryParameters().apply {
+                maxFeatures = 1
+                whereClause = "1 = 1"
+                // find and sort out the orderByFields by most recent first
+                orderByFields.add(
+                    OrderBy(
+                        dateCreatedFieldName,
+                        sortOrder = SortOrder.Descending
+                    )
+                )
+            }
+            lifecycleScope.launch {
+                positioningFeatureTable.queryFeatures(queryParameters)
+                    .onSuccess { queryResults ->
+                        // perform search query using the queryParameters
+                        queryResults.forEach { feature ->
+                            // check if serviceFeatureTable contains positioning data
+                            // The ID that identifies a row in the positioning table.
+                            val globalID =
+                                feature.attributes[positioningFeatureTable.globalIdField].toString()
+                            val positioningId = Guid(globalID)
+                            // Setting up IndoorsLocationDataSource with positioning, pathways tables and positioning ID.
+                            // positioningTable - the "ips_positioning" feature table from an IPS-enabled map.
+                            // pathwaysTable - An ArcGISFeatureTable that contains pathways as per the ArcGIS Indoors Information Model.
+                            // levelsTable - An ArcGISFeatureTable that contains floor levels in accordance with the ArcGIS Indoors Information Model.
+                            // Setting this property enables path snapping of locations provided by the IndoorsLocationDataSource.
+                            // positioningID - an ID which identifies a specific row in the positioningTable that should be used for setting up IPS.
+                            mIndoorsLocationDataSource = IndoorsLocationDataSource(
+                                positioningFeatureTable,
+                                getFeatureTable("Pathways"),
+                                getFeatureTable("levels"),
```

Comment:
```suggestion
                                getFeatureTable("Levels"),
```

This must be case sensitive I think? I was seeing an error - `levels table not found`

Reply:
I had it as `levels` before updating this code, and it was working, in the way, it updated the levels as I moved up and down the floors. Did you test this today, with the updated code?

Reply:
Updated